style/preview.less

```diff
@@ -352,3 +333,63 @@
 	flex-direction: column;
 	justify-content: center;
 }
+
+@media (prefers-color-scheme: dark) {
+	.wikipediapreview {
+		.wikipediapreview-dark-theme();
+	}
+}
+
+@media (prefers-color-scheme: light) {
+	.wikipediapreview {
+		.wikipediapreview-light-theme();
+	}
+}
+
+.wikipediapreview.wikipediapreview-dark-theme {
+	.wikipediapreview-dark-theme();
+}
+
+.wikipediapreview.wikipediapreview-light-theme {
+	.wikipediapreview-light-theme();
+}
+
+.wikipediapreview-dark-theme() {
+	background-color: #202122;
+
+	.wikipediapreview-body {
+		&-message,
+		p {
+			color: #eaecf0;
+		}
+	}
+
+	.wikipediapreview-footer-loading {
+		background-color: #202122;
+	}
+
+	.wikipediapreview-header-wordmark,
+	.wikipediapreview-header-closebtn {
+		filter: invert(1);
+	}
+}
+
+.wikipediapreview-light-theme() {
+	background-color: #fff;
+
+	.wikipediapreview-body {
+		&-message,
+		p {
+			color: #202122;
+		}
+	}
+
+	.wikipediapreview-footer-loading {
+		background-color: #eaecf0;
+	}
+
+	.wikipediapreview-header-wordmark,
+	.wikipediapreview-header-closebtn {
+		filter: unset;
+	}
+}
```

Comment:
Some of the duplication with less mixins could be avoided with CSS custom properties (are they okay to use in this project?). Example:
```css
/* Define default light theme custom properties */
:root {
  --background-color: #ffffff;
  --text-color: #000000;
}

/* Override custom properties for dark theme based on media query preference */
@media (prefers-color-scheme: dark) {
  :root {
    --background-color: #000000;
    --text-color: #ffffff;
  }
}

/* Explicit class for light theme */
.light-theme {
  --background-color: #ffffff;
  --text-color: #000000;
}

/* Explicit class for dark theme */
.dark-theme {
  --background-color: #000000;
  --text-color: #ffffff;
}

/* Use custom properties */
body {
  background-color: var(--background-color);
  color: var(--text-color);
}
```

Reply:
I don't see why it wouldn't be okay to use them, I refactored to use CSS custom properties and indeed it seems a bit cleaner (this way seems to be more of a standard - https://web.dev/articles/prefers-color-scheme#stylesheet_architecture), thanks

Reply:
`:root` represent the whole document, probably the `<html>` or `<body>` tag. It's important to remember that this is a small component hosted within any website. If we want to use an approach like that, everything has to be prefixed so that even if it bleeds into the host, it has no side effects on it.

Reply:
[LESS variables](https://lesscss.org/features/#variables-feature) would probably give you a similar coding style but with actual isolation.

Reply:
Oh that's a good point, I can update using LESS variables and maybe prefix it anyway as something like `--wikipediapreview-primary-background-color`

Reply:
is variable (https://lesscss.org/features/#variables-feature) helpful here?

Reply:
Testing out it out [like so](https://github.com/wikimedia/wikipedia-preview/commit/b6c0b1ce50b25e5a736a30ff3a591e0ca4dbf26f) is not working properly, not overriding/updating variables as expected. GPT said "no, you cannot update LESS variables inside a media query. LESS variables are compiled into static CSS values during the compilation process, and they cannot be changed dynamically at runtime, including within media queries". Does that sound like hallucination to you? 

In any case, we can also use well prefixed CSS custom vars or LESS mixins, we have options

Reply:
Leaving it [with prefixes](https://github.com/wikimedia/wikipedia-preview/pull/186/commits/06edf0ff68438978beb596dd7edafbfb01f113ef) for now, let's make a decision next week

Reply:
this looks good now i think, but can we still use mixin() here so we don't repeat the --var settings?

Reply:
Do we need to repeat the styles for the "light" variants? Aren't those the defaults?

Reply:
can we also write in README to explain how the site owner custom their dark mode styling?

custom code like this, and explain what's the var there
```
@media (prefers-color-scheme: dark) {
	.wikipediapreview {
		--wikipediapreview-primary-background-color: #202122;
	        --wikipediapreview-secondary-background-color: #202122;
		--wikipediapreview-primary-color: #eaecf0;
		--wikipediapreview-filter-setting: invert(1);
	}
}
```

Reply:
Great feedback all, thanks